KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/TermComponentQuery.java

```diff
@@ -165,47 +175,59 @@ public String getQueryString() {
 
     @Override
     public KeywordCachedArtifact writeSingleFileHitsToBlackBoard(String termHit, KeywordHit hit, String snippet, String listName) {
-        final String MODULE_NAME = KeywordSearchModuleFactory.getModuleName();
-
-        //there is match actually in this file, create artifact only then
         BlackboardArtifact bba;
-        KeywordCachedArtifact writeResult;
         Collection<BlackboardAttribute> attributes = new ArrayList<>();
         try {
-            bba = hit.getContent().newArtifact(ARTIFACT_TYPE.TSK_KEYWORD_HIT);
-            writeResult = new KeywordCachedArtifact(bba);
-        } catch (Exception e) {
-            logger.log(Level.WARNING, "Error adding bb artifact for keyword hit", e); //NON-NLS
-            return null;
-        }
+            //if the keyword hit matched the  credit card number keyword/regex...
+            if (keyword.getType() == ATTRIBUTE_TYPE.TSK_ACCOUNT_NUMBER) {
+                bba = hit.getContent().newArtifact(ARTIFACT_TYPE.TSK_CREDIT_CARD_ACCOUNT);
+                // make account artifact
+                //try to match it against the track 1 regex
+                Matcher matcher = TRACK1_PATTERN.matcher(hit.getSnippet());
+                if (matcher.find()) {
+                    parseTrackData(bba, matcher, hit, true);
```

Comment:
Please consider eliminating the flag parameter to the parsing method per discussion, thanks.